hydra-node/src/Hydra/HeadLogic.hs

```diff
@@ -560,24 +644,237 @@ onOpenNetworkAckSn Environment{party} openState otherParty snapshotSignature sn
             InvalidMultisignature{multisig = show multisig, vkeys}
 
   maybeEmitSnapshot outcome =
-    if isLeader parameters party nextSn && not (null localTxs)
+    if partyIsLeader && not (null localTxs)
       then
         outcome
           <> newState SnapshotRequestDecided{snapshotNumber = nextSn}
-          <> cause (NetworkEffect $ ReqSn nextSn (txId <$> localTxs))
+          -- Spec Gap: how S⁻.s + 1 is calculated is not the same as nextSn
```

Comment:
Could make it more similar, but line 576 `wait ŝ = s` and 591 (new) `S⁻ ← snObj(vˆ, ŝ, 𝑈ˆ, T̂ , txω)` implies that it's the same